From 3c250cfe63b7af59ce0141885596a426053fe669 Mon Sep 17 00:00:00 2001
From: Fraser Cormack <fraser@codeplay.com>
Date: Mon, 30 Jan 2023 14:06:50 +0000
Subject: [PATCH] Patch sub-group testing

There's currently a bug in the CTS whereby our degenerate sub-groups
crash the test, which uses too small a fixed data structure to track all
work-items in a work-group larger than 128.

While we wait for an official patch upstream, I've quickly patched it so
the data structure is as big as the largest local work-group size tested
by the tests.
---
 test_conformance/subgroups/subhelpers.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test_conformance/subgroups/subhelpers.h b/test_conformance/subgroups/subhelpers.h
index 8600088e..fafe35fe 100644
--- a/test_conformance/subgroups/subhelpers.h
+++ b/test_conformance/subgroups/subhelpers.h
@@ -29,7 +29,8 @@
 #include <map>
 
 extern MTdata gMTdata;
-typedef std::bitset<128> bs128;
+// See https://github.com/KhronosGroup/OpenCL-CTS/pull/1621
+typedef std::bitset<200> bs128;
 extern cl_half_rounding_mode g_rounding_mode;
 
 bs128 cl_uint4_to_bs128(cl_uint4 v);
-- 
2.34.1


